c7n/resources/asg.py

```diff
@@ -492,6 +492,49 @@ def process(self, asgs, event=None):
         return super(VpcIdFilter, self).process(asgs)
 
 
+@filters.register('failing')
+class FailingFilter(Filter):
+    schema = type_schema(
+        'failing', count={'type': 'number'})
+
+    def asg_failed_launch(self, activities={}, token=None):
```

Comment:
its unclear why these parameters are present, given their never passed. you should never use a mutable variable as a default argument to the method, its always a bad ideas, as mutations persist across invocations since the var is bound to the function definition.